mvc/src/main/java/com/interface21/webmvc/servlet/mvc/tobe/ControllerScanner.java

```diff
@@ -0,0 +1,40 @@
+package com.interface21.webmvc.servlet.mvc.tobe;
+
+import com.interface21.context.stereotype.Controller;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.reflections.Reflections;
+
+public class ControllerScanner {
+
+    private final Reflections reflections;
+
+    public ControllerScanner(final Reflections reflections) {
+        this.reflections = reflections;
+    }
+
+    public Map<Class<?>, Object> getControllers() {
+        final Set<Class<?>> controllersClasses = reflections.getTypesAnnotatedWith(Controller.class);
+        return instantiateControllers(controllersClasses);
+    }
+
+    private Map<Class<?>, Object> instantiateControllers(final Set<Class<?>> controllersClasses) {
+        return controllersClasses.stream()
+                .collect(Collectors.toMap(
+                        controllerClass -> controllerClass,
+                        this::getInstanceBy
+                ));
```

Comment:
```suggestion
        return controllersClasses.stream()
                .collect(Collectors.toMap(
                        Function.identity(),
                        this::getInstanceBy
                ));
```

자기 자신을 반환하는 함수의 경우 `Function.identity()`를 활용할 수 있습니다!

참고: https://juno-juno.tistory.com/103

Reply:
자료 첨부까지! 감사합니다 :)

[refactor: Function.identity()로 가독성 개선](https://github.com/dye0p/java-mvc/commit/f59d4498e39d61a1369a6f9ed5f9cc7c56aee53c)